explorer/interpreter/value.h

```diff
@@ -893,16 +921,25 @@ class ParameterizedEntityName : public Value {
         declaration_(declaration),
         params_(params) {}
 
+  explicit ParameterizedEntityName(Nonnull<const Declaration*> declaration,
+                                   Nonnull<const TuplePattern*> params,
+                                   const std::vector<NamedValue>& alternatives)
+      : Value(Kind::ParameterizedEntityName),
+        declaration_(declaration),
+        params_(params),
+        alternatives_(alternatives) {}
   static auto classof(const Value* value) -> bool {
     return value->kind() == Kind::ParameterizedEntityName;
   }
 
   auto declaration() const -> const Declaration& { return *declaration_; }
   auto params() const -> const TuplePattern& { return *params_; }
+  auto alternatives() const -> std::vector<NamedValue> { return alternatives_; }
 
  private:
   Nonnull<const Declaration*> declaration_;
   Nonnull<const TuplePattern*> params_;
+  std::vector<NamedValue> alternatives_;
```

Comment:
If I understand correctly, `ParameterizedEntityName` represents a _name_. In other words, in an expression like `Optional(i32)`, it represents the `Optional` part. That being the case, it seems really weird for this class to have a list of alternatives as a data member. Could the code that needs the alternatives get them out of `declaration` instead?